test/suites/basic.sh

```diff
@@ -755,6 +755,64 @@ EOF
   lxc config trust remove "${fingerprint}"
 }
 
+test_snap_basic_usage_vm() {
+  lxc launch ubuntu-minimal-daily:24.04 v1 --vm -c limits.memory=384MiB -d "${SMALL_VM_ROOT_DISK}"
+  waitInstanceReady v1
+  lxc list --fast
+
+  echo "==> Check exec operations"
+  lxc exec v1 -- true
+  ! lxc exec v1 -- false || false
+  [ "$(lxc exec v1 -- hostname)" = "v1" ]
+
+  echo "==> Check VM state transitions"
+  [ "$(lxc list -f csv -c s)" = "RUNNING" ]
+  lxc pause v1
+  [ "$(lxc list -f csv -c s)" = "FROZEN" ]
+  ! lxc stop v1 || false
+  [ "$(lxc list -f csv -c s)" = "FROZEN" ]
+  lxc start v1
+  [ "$(lxc list -f csv -c s)" = "RUNNING" ]
+  lxc stop -f v1
+  [ "$(lxc list -f csv -c s)" = "STOPPED" ]
+
+  echo "==> VM with snapshots"
+  lxc snapshot v1
```

Comment:
These name validation checks dont need to be in the snap test as not snap specific. Please can they be in the basic non snap VM test?